src/blueprint/from_config/operators/http.rs

```diff
@@ -30,6 +30,28 @@ impl<'a> MustachePartsValidator<'a> {
   fn new(type_of: &'a config::Type, config: &'a Config, field: &'a FieldDefinition) -> Self {
     Self { type_of, config, field }
   }
+  fn get_nested_type(&self, tail: &[String], mut type_of: &'a config::Type) -> Result<(Type, String), String> {
+    let mut len = tail.len();
+    for item in tail {
+      let val_type = get_value_type(type_of, item).ok_or_else(|| format!("no value '{}' found", item))?;
+
+      if len == 1 {
+        return Ok((val_type, item.to_owned()));
+      } else if val_type.is_nullable() {
+        return Err(format!("value '{}' is a nullable type", item.as_str()));
+      }
+
+      if let Some(field) = type_of.fields.get(item) {
+        if let Some(_type_of) = self.config.find_type(&field.type_of) {
+          type_of = _type_of;
+        }
+      }
```

Comment:
this is not obvious actually that `type_of` will be updated although in practice it should. Because the first condition is checked already in `get_value_type` and the second should be validated prior to this validation. 

Let's actually handle the case when `type_of` is not updated and return error to prevent issues in the future in case some changes will be made to other code that will break that assumptions

Reply:
I've deleted `get_value_type` and now only check once for `type_of.fields.get`. I'm using `ok_or_else` to throw a `Path [{}] does not exist`, but the code never comes to this inexistent path, as not found will not even enter on the if.

If there are any more needed changes, let me know. We should specify a test case to cover it, but I could not reach that code here. It'd be nice get a hand on it.

Reply:
is is a `Path` or more like `Type` in that case?

I think if you'll try to replicate this the validation will fail with another error since it's will be validated out by another code. I think we don't need to care about it here, so it's ok to not have reproduction in tests for this, but our code should be reliable anyway and we should the handle all the cases

Reply:
Changed it to `Type`